packages/device_info/device_info_platform_interface/lib/method_channel_device_info.dart

```diff
@@ -0,0 +1,22 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+import 'device_info_platform_interface.dart';
+
+const MethodChannel _channel = MethodChannel('plugins.flutter.io/device_info');
+
+/// An implementation of [DeviceInfoPlatform] that uses method channels.
+class MethodChannelDeviceInfo extends DeviceInfoPlatform {
+  // Method channel for Android devices
+  Future<Map<String, dynamic>> androidInfo() async {
+    return Map<String, dynamic>.from(
+        await _channel.invokeMethod('getAndroidDeviceInfo'));
```

Comment:
`return AndroidDeviceInfo.fromMap(await _channel.invokeMethod...`

Reply:
Sure sir ! I'm on it.